Source/WebCore/page/IntersectionObserver.cpp

```diff
@@ -579,7 +583,8 @@ void IntersectionObserver::notify()
 bool IntersectionObserver::isReachableFromOpaqueRoots(JSC::AbstractSlotVisitor& visitor) const
 {
     for (auto& target : m_observationTargets) {
-        if (auto* element = target.get(); containsWebCoreOpaqueRoot(visitor, element))
+        SUPPRESS_UNCOUNTED_LOCAL auto* element = target.get();
```

Comment:
@rniwa @geoffreygaren Do we really want to start spreading the `SUPPRESS_UNCOUNTED_LOCAL` use for things that are false positives in the static analyzer that we should be able to address in the near future? Seems unfortunate to make the codebase look worse just because of a static analyzer bug.

Reply:
So this one, I'm not sure if we can call it a bug in the static analyzer. `containsWebCoreOpaqueRoot` calls `Visitor::containsOpaqueRoot`, which in turn calls `didFindOpaqueRoot`, which is a non-inline function. There isn't really a way that static analyzer can detect this is a trivial function unless we hard-code the function name in the static analyzer or something.

Reply:
I'd prefer to explicitly annotate containsWebCoreOpaqueRoot() as trivial / non-deleting. Otherwise we'll have to suppress in hundreds of places. I guess it would require a small feature in the static analyzer to offer that annotation.